lib/vanagon/engine/hardware.rb

```diff
@@ -0,0 +1,64 @@
+require 'vanagon/engine/base'
+require 'json'
+require 'lock_manager'
+
+LOCK_MANAGER_HOST = ENV['LOCK_MANAGER_HOST'] || 'redis'
+LOCK_MANAGER_PORT = ENV['LOCK_MANAGER_PORT'] || 6379
+VANAGON_LOCK_USER = ENV['USER']
+
+
+class Vanagon
+  class Engine
+    # Class to use when building on a hardware device (e.g. AIX, Switch, etc)
+    #
+    class Hardware < Base
+
+      # This method is used to obtain a vm to build upon
+      # For the base class we just return the target that was passed in
+      def select_target
+        @target = node_lock(@build_hosts)
+      end
+
+      # Poll for a lock
+      def polling_lock(host)
+        Vanagon::Driver.logger.info "Polling for a lock on #{host}."
+        @lockman.polling_lock(host, VANAGON_LOCK_USER, "Vanagon automated lock")
+        Vanagon::Driver.logger.info "Lock acquired on #{host}."
+        puts "Lock acquired on #{host} for #{VANAGON_LOCK_USER}."
+        host
+      end
+
+      # Iterarte over the options and find a node open to lock.
+      def node_lock(hosts)
+        hosts.each do |h|
+          Vanagon::Driver.logger.info "Attempting  to lock #{h}."
+          if @lockman.lock(h, VANAGON_LOCK_USER, "Vanagon automated lock")
+            Vanagon::Driver.logger.info "Lock acquired on #{h}."
+            puts "Lock acquired on #{h} for #{VANAGON_LOCK_USER}."
+            return h
+          end
+        end
+        # If they are all locked, fall back to a polling lock on last item
+        polling_lock(hosts.pop)
```

Comment:
@stahnma is there a specific reason that the function only passes once? it seems like it might be better to just loop around and keep trying them all again and again.